server/src/main/java/org/elasticsearch/repositories/RepositoriesStats.java

```diff
@@ -0,0 +1,112 @@
+        } else {
+            repositoryThrottlingStats = new HashMap<>();
+        }
+    }
+
+    public RepositoriesStats(Map<String, ThrottlingStats> repositoryThrottlingStats) {
+        this.repositoryThrottlingStats = new HashMap<>(repositoryThrottlingStats);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_500_010)) {
+            out.writeMap(repositoryThrottlingStats, StreamOutput::writeString, (o, v) -> v.writeTo(o));
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field("repository", repositoryThrottlingStats);
+        return builder;
+    }
+
+    public Map<String, ThrottlingStats> getRepositoryThrottlingStats() {
+        return Collections.unmodifiableMap(repositoryThrottlingStats);
+    }
+
+    public static class Builder {
+        private final Map<String, ThrottlingStats> repoThrottlingStats = new HashMap<>();
+
+        public Builder add(String repoName, long totalReadThrottledNanos, long totalWriteThrottledNanos) {
+            repoThrottlingStats.put(repoName, new ThrottlingStats(totalReadThrottledNanos, totalWriteThrottledNanos));
+            return this;
+        }
+
+        public RepositoriesStats build() {
+            return new RepositoriesStats(repoThrottlingStats);
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class ThrottlingStats implements ToXContentObject, Writeable {
+
+        private final long totalReadThrottledNanos;
+        private final long totalWriteThrottledNanos;
+
+        ThrottlingStats(StreamInput in) throws IOException {
+            this.totalReadThrottledNanos = in.readLong();
+            this.totalWriteThrottledNanos = in.readLong();
+        }
+
+        public ThrottlingStats(long totalReadThrottledNanos, long totalWriteThrottledNanos) {
+            this.totalReadThrottledNanos = totalReadThrottledNanos;
+            this.totalWriteThrottledNanos = totalWriteThrottledNanos;
+        }
+
+        public long getTotalReadThrottledNanos() {
+            return totalReadThrottledNanos;
+        }
+
+        public long getTotalWriteThrottledNanos() {
+            return totalWriteThrottledNanos;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field("total_read_throttled_nanos", totalReadThrottledNanos);
+            builder.field("total_write_throttled_nanos", totalWriteThrottledNanos);
+            builder.endObject();
+            return builder;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeLong(totalReadThrottledNanos);
+            out.writeLong(totalWriteThrottledNanos);
```

Comment:
These numbers should always be positive and usually won't be very large, so I think we should prefer `writeVLong` (not that saving a few bytes matters much here).